jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -405,6 +435,29 @@ object XGBoost extends Serializable {
     xgBoostModel.setPredictionCol(predCol)
   }
 
+  private def loadPrevBooster(sparkContext: SparkContext, hdfsTmpPath: String):
+      Booster = {
+    if (hdfsTmpPath == null) {
+      null
+    } else {
+      val fs = FileSystem.get(sparkContext.hadoopConfiguration)
+      val tempPath = new Path(hdfsTmpPath)
+      val versions = fs.listStatus(tempPath).map(_.getPath.getName).collect {
+        case fileName if fileName.endsWith(modelSuffix) => fileName.stripSuffix(modelSuffix).toInt
+      }
+      if (versions.nonEmpty) {
+        val version = versions.max
+        val fullPath = s"$hdfsTmpPath/$version$modelSuffix"
```

Comment:
we should not duplicate the path pattern in two places to facilitate the further modification

Reply:
Good Point.